# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1299341833 18000
#      Sat Mar 05 11:17:13 2011 -0500
# Node ID 5454b687fd14cf7865adec625c0985bb069c9b29
# Parent  8f2056646dbae2877fc8696aca0ae819f0433e9f
colorbar.m: Allow location to be specified as a property.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2011-03-05  Ben Abbott <bpabbott@mac.com>
+
+	* plot/colorbar.m: Allow location to be specified as a property.
+
 2010-03-03  Rik  <octave@nomad.inbox5.com>
 
 	* miscellaneous/bunzip2.m, miscellaneous/bzip2.m,
 	miscellaneous/gunzip.m, miscellaneous/gzip.m, miscellaneous/unpack.m,
 	miscellaneous/untar.m, miscellaneous/unzip.m: Improve docstrings
 
 	* miscellaneous/tar.m, miscellaneous/zip.m: Check for invalid arguments
 	at the top, rather than bottom, of code.
diff --git a/scripts/plot/colorbar.m b/scripts/plot/colorbar.m
--- a/scripts/plot/colorbar.m
+++ b/scripts/plot/colorbar.m
@@ -58,33 +58,34 @@ function h = colorbar (varargin)
   ax = [];
   loc = "eastoutside";
   args = {};
   deleting = false;
 
   i = 1;
   while (i <= nargin)
     arg = varargin {i++};
-
     if (ischar(arg))
       if (strcmpi (arg, "peer"))
         if (i > nargin)
           error ("colorbar: missing axes handle after \"peer\"");
         else
           ax = varargin{i++};
           if (!isscalar (ax) || ! ishandle (ax)
               || ! strcmp (get (ax, "type"), "axes"))
             error ("colorbar: expecting an axes handle following \"peer\"");
           endif
         endif
       elseif (strcmpi (arg, "north") || strcmpi (arg, "south")
               || strcmpi (arg, "east") || strcmpi (arg, "west")
               || strcmpi (arg, "northoutside") || strcmpi (arg, "southoutside")
               || strcmpi (arg, "eastoutside") || strcmpi (arg, "westoutside"))
         loc = tolower (arg);
+      elseif (strcmpi (arg, "location") && i <= nargin)
+        loc = tolower (varargin{i++});
       elseif (strcmpi (arg, "off") || strcmpi (arg, "none"))
         deleting = true;
       else
         args{end+1} = arg;
       endif
     else
       args{end+1} = arg;
     endif
